server/server-kv.c

```diff
@@ -46,21 +92,26 @@ handle_request(void *ptr)
 		pthread_mutex_unlock(&mutex);
                 clientfd = node->fd;
 
-                //for test
-                bytes_read = read(clientfd, message, 256);
-		if(bytes_read < 0) {
-			perror("ERROR reading socket");
-		}
-		else if (bytes_read == 0){
-			printf("Client disconnected");
-		}
-		else {
-                        strcpy(message, gwkv_handle_operation(ht, message));
-                        bytes_write = strlen(message)+1;
-			write(clientfd, message, bytes_write);
-                        printf("thread: %d, received:%s\n", pthread_self(), message);
+		while(1){
```

Comment:
It is not a infinite loop, it stops at the point that "curr_char == '\n'";
pls see several lines downward, there is a "if (curr_char == '\n') break;" ,  it jumps out of the loop when "\n" occurs. 

Using similar pattern idea of python loop, such as "for item in lists: "